### PR TITLE
add support for topologySpreadConstraints in deployments

### PR DESCRIPTION
**Pull request by jhohertz**
Signed-off-by: Joe Hohertz <joe@viafoura.com>

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@jhohertz could you please fix the conflict?

**Comment by pierluigilenoci** [comment]
@desaintmartin could you please take a look?

**Review by pierluigilenoci — DISMISSED** [review]

**Review by pierluigilenoci — DISMISSED** [review]

**Comment by pierluigilenoci** [comment]
@jhohertz could you please rebase?

**Comment by pierluigilenoci** [comment]
@desaintmartin could you please take a look?

File: helm/oauth2-proxy/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: oauth2-proxy
-version: 6.2.7
+version: 6.2.8
```
Comment (desaintmartin) [inline helm/oauth2-proxy/Chart.yaml:2]:
```suggestion
version: 6.3.0
```
(Or whatever needed version after rebase): this is new feature according to semver.

**Review by pierluigilenoci — APPROVED** [review]

**Comment by pierluigilenoci** [comment]
@desaintmartin I did the bump. 
If that was the only observation I'd say we're ready for the merge.

**Pull request merged**